codec-http/src/main/java/io/netty/handler/codec/http/QueryStringDecoder.java

```diff
@@ -330,7 +331,10 @@ private static String decodeComponent(String s, int from, int toExcluded, Charse
 
             byteBuf.clear();
             do {
-                byteBuf.put(decodeHexByte(s, i, toExcluded));
+                if (i + 3 > toExcluded) {
+                    throw new IllegalArgumentException("unterminated escape sequence at index " + i + " of: " + s);
+                }
+                byteBuf.put(decodeHexDump(s, i + 1, 2));
```

Comment:
> Use ByteBuf#nioBuffer() will be too expensive. It smells of overengineering )

Perhaps ... if it is too much I'm not against directly adding a `ByteBuffer` utility method.